app/services/EssttpBarsService.scala

```diff
@@ -50,22 +50,22 @@ class EssttpBarsService @Inject() (
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(requestHeader)
 
     val taxId = journey match {
-      case account: Stages.ChosenTypeOfBankAccount     => account.taxId
-      case details: Stages.EnteredDirectDebitDetails   => details.taxId
-      case details: Stages.ConfirmedDirectDebitDetails => details.taxId
-      case conditions: Stages.AgreedTermsAndConditions => conditions.taxId
-      case arrangement: Stages.SubmittedArrangement    => arrangement.taxId
+      case j: AfterComputedTaxId => j.taxId
+      case _                     => throw new RuntimeException("Expected to find a taxId but none found")
     }
```

Comment:
Sorry can this be
```
case _: BeforeComputedTaxId  => throw new RuntimeException("Expected to find a taxId but none found")
case j: AfterComputedTaxId => j.taxId
```
just incase someone ever forgets to use the marking traits on new stages

Reply:
doesn't compile (fruiteless type test)